proto/internal/temporal/server/api/matchingservice/v1/service.proto

```diff
@@ -89,6 +89,18 @@ service MatchingService {
     // Fetch user data for a task queue, this request should always be routed to the node holding the root partition of the workflow task queue.
     rpc GetTaskQueueUserData (GetTaskQueueUserDataRequest) returns (GetTaskQueueUserDataResponse) {}
 
+    // Allows updating the Build ID assignment and redirect rules for a given Task Queue.
+    // (-- api-linter: core::0127::http-annotation=disabled
+    //     aip.dev/not-precedent: We do yet expose versioning API to HTTP. --)
+    rpc UpdateWorkerVersioningRules (UpdateWorkerVersioningRulesRequest) returns (UpdateWorkerVersioningRulesResponse) {}
```

Comment:
I think we should define these request and response types again in `proto/internal/temporal/server/api/matchingservice/v1/request_response.proto`.
The new request/response types can be merely wrappers around the public API's ones. later we may want to add more things in them for internal use.